src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/InteropServices/CSharpDisableRuntimeMarshalling.Fixer.cs

```diff
@@ -0,0 +1,296 @@
+                        IdentifierGenerator identifierGenerator;
+                        if (block is null)
+                        {
+                            identifierGenerator = new IdentifierGenerator(editor.SemanticModel, node.SpanStart);
+                        }
+                        else if (!scopeMap.TryGetValue(block, out identifierGenerator))
+                        {
+                            identifierGenerator = scopeMap[block] = new IdentifierGenerator(editor.SemanticModel, block);
+                        }
+                        if (TryRewriteMethodCall(node, editor, identifierGenerator, addRenameAnnotation: false, fixAllContext.CancellationToken))
+                        {
+                            AddUnsafeModifierToEnclosingMethod(editor, node);
+                        }
+                    }
+                }
+                return editor.GetChangedRoot();
+            }
+        }
+        private class IdentifierGenerator
+        {
+            private int? _nextIdentifier;
+
+            public IdentifierGenerator(SemanticModel model, int offsetForSpeculativeSymbolResolution)
+            {
+                _nextIdentifier = FindFirstUnusedIdentifierIndex(model, offsetForSpeculativeSymbolResolution, "ptr");
+            }
+            public IdentifierGenerator(SemanticModel model, IBlockOperation block)
+            {
+                _nextIdentifier = FindFirstUnusedIdentifierIndex(model, block.Syntax.SpanStart, "ptr");
+                HashSet<string> localNames = new HashSet<string>(block.Locals.Select(x => x.Name));
+                string? identifier = NextIdentifier();
+                while (identifier is not null && localNames.Contains(identifier))
+                {
+                    identifier = NextIdentifier();
+                }
+                if (identifier is not null)
+                {
+                    // The last identifier was not in use, so go back one to use it the next call.
+                    _nextIdentifier--;
+                }
+            }
+
+            public string? NextIdentifier()
+            {
+                if (_nextIdentifier == null || _nextIdentifier == int.MaxValue)
+                {
+                    return null;
+                }
+
+                if (_nextIdentifier == 0)
+                {
+                    _nextIdentifier++;
+                    return "ptr";
+                }
+
+                return $"ptr{_nextIdentifier++}";
+            }
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(DisableRuntimeMarshallingAnalyzer.MethodUsesRuntimeMarshallingEvenWhenMarshallingDisabledId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            return CustomFixAllProvider.Instance;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
```

Comment:
Nit: Can we move the overrides to the top of the file and the nested classes to separate files?